### PR TITLE
feat: Delete on multiselection - EXO-62360

### PR DESCRIPTION
**Pull request by mkrout**

### CODE REVIEW

File: documents-services/src/main/java/org/exoplatform/documents/rest/DocumentFileRest.java [hunk 1]
```diff
@@ -537,6 +568,34 @@ public Response deleteDocument(@Parameter(description = "Document id", required
     }
   }
 
+  @DELETE
+  @Path("bulk/{actionId}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @Operation(summary = "Delete document", method = "DELETE", description = "This deletes document")
```
Comment (hakermi) [inline documents-services/src/main/java/org/exoplatform/documents/rest/DocumentFileRest.java]:
```suggestion
  @Operation(summary = "Delete list of documents", method = "DELETE", description = "This deletes a list of documents")
```

File: documents-services/src/main/java/org/exoplatform/documents/rest/DocumentFileRest.java [hunk 2]
```diff
@@ -537,6 +568,34 @@ public Response deleteDocument(@Parameter(description = "Document id", required
...
+      return Response.status(Status.BAD_REQUEST).entity("documents list is mandatory").build();
+    }
+    long userIdentityId = RestUtils.getCurrentUserIdentityId(identityManager);
+    try {
+      documentFileService.deleteDocuments(actionId, EntityBuilder.toAbstractNodes(documents), userIdentityId);
+      return Response.ok().build();
+    } catch (IllegalAccessException e) {
+      return Response.status(Status.UNAUTHORIZED).entity(e.getMessage()).build();
+    } catch (Exception e) {
+      LOG.error("Error when deleting the news target with name " + userIdentityId, e);
```
Comment (hakermi) [inline documents-services/src/main/java/org/exoplatform/documents/rest/DocumentFileRest.java]:
```suggestion
      LOG.error("Error while deleting documents", e);
```

File: documents-api/src/main/java/org/exoplatform/documents/storage/JCRDeleteFileStorage.java [hunk 3]
```diff
@@ -45,4 +51,34 @@ public interface JCRDeleteFileStorage {
...
+   * Delete document (Move to trash)
+   *
+   * @param session
+   * @param folderPath
+   * @param documentId
+   * @param favorite
+   * @param checkToMoveToTrash
+   * @param delay
+   * @param identity
+   * @param userIdentityId
```
Comment (hakermi) [inline documents-api/src/main/java/org/exoplatform/documents/storage/JCRDeleteFileStorage.java]:
```suggestion
   * @param session current session
   * @param folderPath folder path
   * @param documentId document id
   * @param favorite favorite or not
   * @param checkToMoveToTrash check whether to move to trash
   * @param delay waiting delay
   * @param identity user identity
   * @param userIdentityId user identity id
```

File: documents-api/src/main/java/org/exoplatform/documents/storage/JCRDeleteFileStorage.java [hunk 4]
```diff
@@ -45,4 +51,34 @@ public interface JCRDeleteFileStorage {
...
+                      Identity identity,
+                      long userIdentityId) throws ObjectNotFoundException, RepositoryException;
+
+  /**
+   * Delete a list of document (Move to trash)
+   *
+   * @param actionId
+   * @param items
+   * @param identity
+   * @param authenticatedUserId
```
Comment (hakermi) [inline documents-api/src/main/java/org/exoplatform/documents/storage/JCRDeleteFileStorage.java]:
```suggestion
   * @param actionId action id
   * @param items list of items to delete
   * @param identity user identity
   * @param authenticatedUserId current authenticated user id 
```

File: documents-services/src/main/java/org/exoplatform/documents/rest/DocumentFileRest.java [hunk 5]
```diff
@@ -74,18 +79,44 @@ public class DocumentFileRest implements ResourceContainer {
...
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @Path("/settings")
+  @Operation(summary = "Get User documents settings", method = "GET")
+  @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+      @ApiResponse(responseCode = "401", description = "Unauthorized operation"),
+      @ApiResponse(responseCode = "500", description = "Internal server error"), })
```
Comment (hakermi) [inline documents-services/src/main/java/org/exoplatform/documents/rest/DocumentFileRest.java]:
```suggestion
  @ApiResponses(value = { 
       @ApiResponse(responseCode = "200", description = "Request fulfilled"),
       @ApiResponse(responseCode = "500", description = "Internal server error"), })
```

File: documents-storage-jcr/src/test/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtilTest.java [hunk 6]
```diff
@@ -304,7 +304,6 @@ public void retrieveFileContentProperties() throws RepositoryException, IOExcept
 
   @Test
   public void isValidDocumentTitle() {
-    assertFalse(JCRDocumentsUtil.isValidDocumentTitle("test:<*?"));
```
Comment (hakermi) [inline documents-storage-jcr/src/test/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtilTest.java:307]: Why this is removed ?

**Review by hakermi — CHANGES_REQUESTED** [review]

File: documents-storage-jcr/src/test/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtilTest.java (hunk 6 above)
Comment (mkrout) [reply documents-storage-jcr/src/test/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtilTest.java:307]: This test never pass on windows

**Review by hakermi — APPROVED** [review]

**Pull request merged**